### PR TITLE
Use Glean Annotations to get app logos and tags

### PR DESCRIPTION
**Pull request by wlach**
This replaces a bunch of hardcoded logic inside the Glean Dictionary
and gives us more flexibility in the future.
Depends on https://github.com/mozilla/glean-annotations/pull/98

### CODE REVIEW

File: etl/glean_etl.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 # Various additional sources of metadata
 ANNOTATIONS_URL = os.getenv(
-    "ANNOTATIONS_URL", "https://mozilla.github.io/glean-annotations/api.json"
+    "ANNOTATIONS_URL", "https://deploy-preview-98--glean-annotations.netlify.app/api.json"
```
Comment (wlach) [inline etl/glean_etl.py:17]: This needs to be reverted before this lands, added just for testing

**Comment by wlach** [comment]
Pretty happy with how this came out: 
image

File: etl/glean_etl.py [hunk 2]
```diff
@@ -59,9 +59,25 @@ def _get_annotation(annotations_index, origin, item_type, identifier=None):
...
+            incorporated.update(
+                {"logo": f"/data/{item['app_name']}/" + _get_logo_filename(item_annotation["logo"])}
+            )
+
+        if item_annotation.get("featured"):
+            incorporated["featured"] = True
+
+        # we use the `app_tags` property to disambiguate between the tags
+        # that are a property of an application vs. the tags that it has
+        # on other things
```
Comment (wlach) [inline etl/glean_etl.py:75]: I realize all of this logic is kind of confusing, but I'm not sure how to make it better right now.

**Review by Iinh — APPROVED** [review]
this looks so good! I love the newly updated logos 🎉

**Pull request merged**